### PR TITLE
Feat: Add Search Files endpoint to v1

### PR DESCRIPTION
**Pull request by skairunner**
Adds search endpoint to v1, because they currently only exist for v0 in the form of "get" and "list". The return type File is already thin, with only metadata, name, id, extid, so no modifications are needed.

### CODE REVIEW

**Comment by skairunner** [comment]
Am going to be making one change: instead of wrapping the v0 API, I'll have the v1 get/list methods call the search API instead.

**Review by amorken — APPROVED** [review]

File: cognite/seismic/protos/v1/seismic_service.proto [hunk 1]
```diff
@@ -107,4 +107,6 @@ service SeismicAPI {
     Can retrieve a full file or create a new cropped file filtering on areas of interest only.
     **/
     rpc GetSegYFile (SegYSeismicRequest) returns (stream SegYSeismicResponse) {}
+
+    rpc SearchFiles(SearchFilesRequest) returns (stream File) {}
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service.proto:111]: Oh no we forgot a doc comment...

File: cognite/seismic/protos/v1/seismic_service_datatypes.proto [hunk 2]
```diff
@@ -133,7 +132,7 @@ Can search by id, name, or prefix.
 message SearchSpec {
     oneof findby {
         int64 id = 1;
-        string id_string = 6; // Surveys must have string format ids. Other objects can't use this.
+        string id_string = 6; // Surveys and files must have string format ids. Other objects can't use this.
```
Comment (amorken) [inline cognite/seismic/protos/v1/seismic_service_datatypes.proto:135]: Hm, could we change this to name to uuid while we have the chance?

File: cognite/seismic/protos/v1/seismic_service.proto (hunk 1 above)
Comment (amorken) [reply cognite/seismic/protos/v1/seismic_service.proto:111]: Oooh, I see another problem. Looking at the v0 GetFile endpoint, there's a GetFileResponse message wrapping File protobufs. All the custom headers and stuff are in the GetFileResponse protobuf. This is... unfortunate.

But now that we're redoing the API in v1 we can fix that!  :grinning:

**Pull request merged**